service/src/main/java/org/broadinstitute/listener/relay/inspectors/SetDateAccessedInspector.java

```diff
@@ -0,0 +1,154 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.google.common.base.Strings;
+import com.microsoft.azure.relay.RelayedHttpListenerRequest;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.Locale;
+import org.apache.http.client.utils.URIBuilder;
+import org.broadinstitute.listener.relay.Utils;
+import org.broadinstitute.listener.relay.inspectors.InspectorType.InspectorNameConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component(InspectorNameConstants.SET_DATE_ACCESSED)
+public class SetDateAccessedInspector implements RequestInspector {
```

Comment:
Quick question @giventocode, does `SetDateAccessedInspector` _always_ run in the listener, meaning the new config options are required?

Currently the listener is deployed in front of runtimes and apps; I think at the moment we only want to call `setDateAccessed` for runtime deployments since that is where we have [auto-]pause implemented. We probably don't want apps to be calling `/api/v2/runtimes/%s/%s/setDateAccessed` (in the future we may want to auto-pause apps as well).

Would it be possible for Leo to enable/disable `SetDateAccessedInspector` as needed at deployment time?

Reply:
Yeah, good question. The inspectors can be turn on/off via config. I updated the properties with the new properties for the `setDateAccessed` inspector, but with the inspector **disabled** .

https://github.com/DataBiosphere/terra-azure-relay-listeners/blob/main/service/src/main/resources/application.yml

```yaml
  requestInspectors:
#    - headersLogger
    - samChecker
#    - setDateAccessed
```

Reply:
Oh gotcha.. missed that, thank you! 

(Maybe at some point we can refactor this to use events instead of direct listener->Leo coupling ;))